packages/diagrams/lib/src/future_builder.dart

```diff
@@ -0,0 +1,106 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+import 'dart:io';
+
+import 'package:diagram_capture/diagram_capture.dart';
+import 'package:flutter/material.dart';
+
+import 'diagram_step.dart';
+
+const String _futureBuilder = 'future_builder';
+const String _futureBuilderError = 'future_builder_error';
+const Duration _futureDuration = Duration(seconds: 1);
+const Duration _pauseDuration = Duration(seconds: 1);
+final Duration _totalDuration = _futureDuration + _pauseDuration;
+
+class FutureBuilderDiagram extends StatefulWidget implements DiagramMetadata {
```

Comment:
Let's also make note to update the sample code in the follow-up framework PR so this is a runnable sample app.